### ISSUE TITLE
Get mutated argument after function call

### ISSUE THREAD

**Issue by dwoznicki** [napi-rs/napi-rs#436]
I have a `JsObject` created in Rust that I pass in as an argument to `JsFunction.call`. The function mutates the object in JS context. Afterwards, I want to read the mutated object and see what's changed.

Here's the sample code:

```javascript
class XPlugin {
    apply(obj) {
        obj.num = 5;
    }
}

module.exports = {
    plugins: [
        new XPlugin(),
    ]
};
```

```rust
#[js_function(1)]
pub fn bundle(cx: CallContext) -> napi::Result<JsUndefined> {
    let jsconf = cx.get::<JsObject>(0)?; // the module.exports object from above
    if jsconf.has_named_property("plugins")? {
        let plugins = jsconf.get_named_property::<JsArrayBuffer>("plugins")?;
        for i in 0..plugins.get_array_length()? {
            let mut obj = cx.env.create_object()?;
            obj.set_named_property("num", cx.env.create_int32(4)?)?;

            let plugin = plugins.get_element::<JsObject>(i)?;
            let apply_func = plugin.get_named_property::<JsFunction>("apply")?;
            apply_func.call(None, &[obj.into_unknown()])?;

            // Here I want to get the updated obj.num value and print it or something.
        }
    }
    cx.env.get_undefined()
}
```

Is there any way to accomplish this?

**Comment by Brooooooklyn** [comment]
Too busy these days, I will reply soon

**Comment by dwoznicki** [comment]
Gotcha. Thanks for the reply.

**Comment by Brooooooklyn** [comment]
After https://github.com/napi-rs/napi-rs/pull/524, you can call `apply_func.call(None, &[&obj])?;`. The `call` method will not take the ownership of `arguments`, is that what you want?  @kdy1 @dwoznicki

**Comment by kdy1** [comment]
@Brooooooklyn I think it's enough. Thank you a lot!

**Comment by Brooooooklyn** [comment]
Published `napi@1.4.0` and `napi-derive@1.1.0`
